### ISSUE TITLE
Batch delete files in a conversation or delete all files in the conversation when it is deleted.

### ISSUE THREAD

**Issue by DareDareHub** [nextcloud/spreed#6469]
Sometimes conversations need to be deleted.
Problem is, images / files remain on the server when a conversation is deleted, so one must delete the files one by one before deleting the conversation.

It would be nice to have a feature to either,

1.  list all image / files in a conversation, to be able to delete them all at once in one click (a bit like "Scrapy APP" in Slack)
or
2. when a conversation is deleted, to simply have an option with a checkbox to delete all images / files.

(this should be a feature made for the admins that should allow them to free space on the server)
Cheers

**Comment by xcarcelle** [comment]
@DareDareHub thanks for raising this feature and @nickvergessen thanks for adding this idea to backlog list.
One comment/idea : would it be difficult to store each conversations "added-to-the-discussion" files into separated directories (named per conversation title) in the talk directory. So the user could easily find the files of one conversation and manipulate those files (for instance delete all files from a conversation) ? Cheers

**Comment by DareDareHub** [comment]
@xcarcelle that's a great idea, sounds clean and easy to implement.

**Comment by DareDareHub** [comment]
Yes, the more I think about it this, the more I think it would be the best idea :

To simply have separate folders for each conversation in the Talk folder, so that all attached files are tidy and can be deleted easily.

**Comment by nickvergessen** [comment]
> To simply have separate folders for each conversation in the Talk folder, so that all attached files are tidy and can be deleted easily.

Doesn't really easily work. Conversation names are not unique, so they would need to use the technical identifier (part after `/call/` in the url) which then once deleted doesn't help the user anymore to find it and delete it. Also when a conversation is then renamed, the directory would have to be renamed too, but we can't do that fully automatic as we don't know if a user manually changed the path and moved it somewhere else

**Comment by DareDareHub** [comment]
I understand, 
the way I saw it was a bit like it works in slack, a channel cannot be renamed, the "uploaded" files (not the "shared" files) should be avilable to delete. 

I don't know if you can imagine the work it takes to close a conversation and to clean all the files that go with it.
The "shared" files disappear without problem but the "uploaded" files remain in each user's Talk folder. 
In my opinion these "uploaded" files should be deletable by the admin no matter who posted them, these files are not part of the "cloud system" they are just here for a while in a conversation.

**Comment by nickvergessen** [comment]
The thing is that plan only works for files which are in that user's Talk/ folder. However there is no programmatically way to detect this in all setup we support.

Also it depends on the way how you use it. E.g. I mostly upload the file to a strategic position and share it from there into a chat rather than simply dropping it into the chat.

**Comment by DareDareHub** [comment]
Yes that is a great idea, 

"I mostly upload the file to a strategic position and share it from there into a chat rather than simply dropping it into the chat."

and this workflow is something great compared to what other chat tools propose.

Nevertheless when you are for example out of the office, taking a picture for the team or simply sharing a "quick" document found on the web, or a screenshot, these are not always files you need to put in a strategic position in the beginning. 
On the contrary you'd want to not care about them for a while, and then when in need of space on your server or simply when cleaning up the conversations, you would like all that "junk" to be easily disposable.
